frontend/archive/src/main/java/org/pytorch/serve/archive/model/ModelArchive.java

```diff
@@ -80,15 +80,28 @@ public static ModelArchive downloadModel(
             }
         }
 
-        if (new File(url).isDirectory()) {
+        File directory = new File(url);
+        if (directory.isDirectory()) {
             // handle the case that the input url is a directory.
             // the input of url is "/xxx/model_store/modelXXX" or
             // "xxxx/yyyyy/modelXXX".
-            return load(url, new File(url), false);
+            File[] fileList = directory.listFiles();
```

Comment:
So, does it work only with `/x/model_store/modely` ?

Is `/a/b/model_store/modelyyy' valid?

Reply:
either case can work.